lib/utils.js

```diff
@@ -109,28 +117,37 @@ function requestWithRetry(request, options, callback) {
  * the details of working with auth in the client code.
  *
  * @param {Array<string>} scopes list of scopes to request as part of auth
+ * @param {Object} config an object with extra configuration parameters (such
+ *     as keyFile or key)
  * @return {function(Object, function(=?,=?,=?):?)} request style function
  *     accepting (options, callback)
  */
-function authorizedRequestFactory(scopes) {
-
+function authorizedRequestFactory(scopes, config) {
+  var authClient;
```

Comment:
One approach to this would be to make the call to `getAuthClient` outside of `makeRequest` as it only requires the `scopes` and `config`. The case where the `authClient` is not yet initialized must still be handled inside `makeRequest` but that will increase the chance that it is available for the first call to `makeRequest`. Making things event based so that requests can be queued until an auth client is ready sounds like a good approach to me.